src/main/java/bossmonster/controller/GameController.java

```diff
@@ -0,0 +1,130 @@
+import bossmonster.domain.player.Skill;
+import bossmonster.domain.player.PlayerName;
+import bossmonster.domain.player.PlayerStats;
+import bossmonster.view.InputView;
+import bossmonster.view.OutputView;
+
+
+public class GameController {
+
+    InputView inputView;
+    OutputView outputView;
+    MonsterGame monsterGame;
+
+    public GameController(InputView inputView, OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+    }
+
+    public void run() {
+        init();
+        start();
+        play();
+        printResult();
+    }
+    private void init() {
+        Monster monster = createBossMonster();
+        Player player = createPlayer();
+        monsterGame = new MonsterGame(monster, player);
+    }
+
+    private void start() {
+        printStart();
+        printPlayerAndMonsterInfo();
+        monsterGame.start();
+    }
+    private void printStart() {
+        outputView.printStart();
+    }
+
+    private void play() {
+        while (monsterGame.isRun()) {
+            proceedPlayerTurn();
+            proceedBossMonsterTurn();
+            printPlayerAndMonsterInfo();
+        }
+    }
+    private void proceedPlayerTurn() {
+        while (true) {
+            try {
+                String skillNo = inputView.askSkillNo();
+                Skill skill = monsterGame.attackMonster(skillNo);
+                outputView.printPlayerAttack(skill);
+                return;
+            } catch (IllegalArgumentException e) {
+                outputView.printErrorMessage(e);
+            }
+        }
+    }
+
+    private void proceedBossMonsterTurn() {
+        if (monsterGame.isMonsterAlive()) {
+            int bossDamage = monsterGame.attackPlayer();
+            outputView.printMonsterAttack(bossDamage);
+        }
+    }
+
+    private void printPlayerAndMonsterInfo() {
+        if (monsterGame.isMonsterAlive()) {
+            outputView.printMonsterInfo(monsterGame.getMonster());
+            outputView.printMonsterCharacter(monsterGame.getGameStatus());
+            outputView.printPlayerInfo(monsterGame.getPlayer());
+        }
+    }
+
+    private void printResult() {
+        Player player = monsterGame.getPlayer();
+        if (monsterGame.isPlayerWin()) {
+            int matchCount = monsterGame.getMatchCount();
+            outputView.printWin(player, matchCount);
+            return;
+        }
+        outputView.printLose(player);
+    }
+
+    private Player createPlayer() {
+        PlayerName playerName = createPlayerName();
+        PlayerStats playerStats = createPlayerStats();
+        return new Player(playerName, playerStats);
+    }
+
+    private PlayerName createPlayerName() {
+        while (true) {
+            try {
+                return new PlayerName(inputView.askPlayerName());
+            } catch (IllegalArgumentException e) {
+                outputView.printErrorMessage(e);
+            }
+        }
+    }
```

Comment:
```suggestion
    private PlayerName createPlayerName() {
        try {
            return new PlayerName(inputView.askPlayerName());
        } catch (IllegalArgumentException e) {
            outputView.printErrorMessage(e);
            return createPlayerName();
        }
    }
    }
```

위 코드처럼 하면 while 문을 사용하지 않아도 될 것 같아요!

Reply:
메서드로 호출하다보면 StackOverflow가 발생된다 생각하여 while true로 구성하였습니다 while 방식이 이상하다고 생각하시는걸까요?